bodhi/tests/server/consumers/test_pungi_wrapper.py

```diff
@@ -0,0 +1,419 @@
+    def test_compose_repo(self, load_variants_config, execute_phases):
+        compose = mock.Mock()
+        variants_conf = mock.Mock()
+
+        wrapper = PungiWrapper(compose, variants_conf)
+
+        wrapper.compose_repo()
+
+        load_variants_config.assert_called_once_with()
+        execute_phases.assert_called_once_with()
+
+    @mock.patch("shutil.copyfileobj")
+    def test_load_variants_config(self, copyfileobj, tmpdir):
+        logger = mock.Mock()
+        topdir = str(tmpdir.mkdir("mash_dir"))
+        compose_id = "f27-modular-updates"
+        pungi_conf_path = get_pungi_conf(tmpdir)
+        pungi_conf = PungiConfig(str(pungi_conf_path), logger)
+        compose_dir = PungiWrapper.init_compose_dir(topdir, pungi_conf, compose_id)
+
+        compose = get_compose(pungi_conf, compose_dir)
+        updates, builds = get_updates_and_builds()
+        variants_conf = VariantsConfig(updates, builds)
+        topdir = compose.topdir
+
+        wrapper = PungiWrapper(compose, variants_conf)
+        wrapper.load_variants_config()
+
+        assert len(wrapper.compose.all_variants)
+        copyfileobj.assert_called_once()
+
+    @mock.patch("pungi.metadata")
+    def test_write_repo_metadata(self, pungi_metadata, tmpdir):
+        logger = mock.Mock()
+        topdir = str(tmpdir.mkdir("mash_dir"))
+        compose_id = "f27-modular-updates"
+        pungi_conf_path = get_pungi_conf(tmpdir)
+        pungi_conf = PungiConfig(str(pungi_conf_path), logger)
+        compose_dir = PungiWrapper.init_compose_dir(topdir, pungi_conf, compose_id)
+
+        compose = get_compose(pungi_conf, compose_dir)
+        updates, builds = get_updates_and_builds()
+        variants_conf = VariantsConfig(updates, builds)
+        topdir = compose.topdir
+
+        wrapper = PungiWrapper(compose, variants_conf)
+        wrapper.load_variants_config()
+        wrapper.write_repo_metadata()
+        assert pungi_metadata.write_tree_info.call_count == 8
+        assert pungi_metadata.write_discinfo.call_count == 8
+        assert pungi_metadata.write_media_repo.call_count == 8
+
+    @mock.patch("os.symlink")
+    @mock.patch("os.unlink")
+    def test_create_latest_symlinks(self, unlink, symlink, tmpdir):
+        logger = mock.Mock()
+        topdir = str(tmpdir.mkdir("mash_dir"))
+        compose_id = "f27-modular-updates"
+        pungi_conf_path = get_pungi_conf(tmpdir)
+        pungi_conf = PungiConfig(str(pungi_conf_path), logger)
+        compose_dir = PungiWrapper.init_compose_dir(topdir, pungi_conf, compose_id)
+
+        compose = get_compose(pungi_conf, compose_dir)
+        updates, builds = get_updates_and_builds()
+        variants_conf = VariantsConfig(updates, builds)
+        topdir = compose.topdir
+
+        wrapper = PungiWrapper(compose, variants_conf)
+        wrapper.create_latest_repo_links()
+        unlink.assert_called_once()
+        symlink.assert_called_once()
```

Comment:
Let's assert what these were called with. In general a lot of these tests needs much stronger assertions so that we know the right behaviors happened.